### PR TITLE
Fluentd plugins ansible playbook

### PR DESCRIPTION
**Pull request by sebsoto**
This commit sets up the logging ansible playbook, and
includes functionality for installing the fluentd plugins
required for OpenShift nodes.

### CODE REVIEW

File: ansible/tasks/logging/main.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+- hosts: win
+  tasks:
+  #- name: Logging | Install fluentd
+  #  include_tasks: install_fluentd.yml
+  #
+  - name: Logging | Install fluentd plugins
+    include_tasks: install_plugins.yml
+
+  #- name: Logging | Configure
+  #  include_tasks: configure.yml
```
Comment (aravindhp) [inline ansible/tasks/logging/main.yml]: What's with the commented out code?

File: ansible/tasks/logging/install_plugins.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      - {name: fluent-plugin-kubernetes_metadata_filter}
+      - {name: fluent-plugin-concat}
+      - {name: fluent-plugin-elasticsearch}
+      - {name: fluent-plugin-multi-format-parser}
+      - {name: fluent-plugin-record-modifier}
+      - {name: fluent-plugin-rewrite-tag-filter}
+      - {name: fluent-plugin-systemd}
+      - {name: fluent-plugin-viaq_data_model}
+      - {name: fluent-plugin-remote-syslog}
+      - {name: fluent-plugin-prometheus}
```
Comment (aravindhp) [inline ansible/tasks/logging/install_plugins.yml]: Is there a source of truth for the list of plugins? If yes, then it should be linked here.

File: ansible/tasks/logging/main.yml (hunk 1 above)
Comment (sebsoto) [reply ansible/tasks/logging/main.yml]: It was meant to be a guideline as to how this piece of code fits in to the greater playbook. I can either add TODOs, or remove the commented out code if needed

File: ansible/tasks/logging/install_plugins.yml (hunk 2 above)
Comment (sebsoto) [reply ansible/tasks/logging/install_plugins.yml]: Added source in a comment

File: tools/ansible/tasks/logging/install_plugins.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+---
+# Installing plugins listed here: https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/Gemfile
```
Comment (ravisantoshgudimetla) [inline tools/ansible/tasks/logging/install_plugins.yml]: I am fine with this for now but going forward, we need to somehow get the list from the repo or the plugins installed on the linux host and get the plugin names from there.

Comment (sebsoto) [reply tools/ansible/tasks/logging/install_plugins.yml]: I agree, we should address this in a future commit

**Comment by ravisantoshgudimetla** [comment]
Why is this having WIP tag?

File: tools/ansible/tasks/logging/test/logging_test.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	"fluent-plugin-rewrite-tag-filter":         "",
+	"fluent-plugin-systemd":                    "",
+	"fluent-plugin-viaq_data_model":            "",
+	"fluent-plugin-remote-syslog":              "",
+	"fluent-plugin-prometheus":                 "",
+}
+
+// TestPluginsInstalled tests that the fluentd plugins were all installed correctly
+func TestPluginsInstalled(t *testing.T) {
+	out, err := exec.Command("fluent-gem", "list").Output()
```
Comment (ravisantoshgudimetla) [inline tools/ansible/tasks/logging/test/logging_test.go]: Please add a comment on where you want the test to be executed.

Comment (sebsoto) [reply tools/ansible/tasks/logging/test/logging_test.go]: :+1:

**Pull request merged**